server/views/media_picker.py

```diff
@@ -148,65 +154,74 @@ def _cached_featured_collection_list(tag_id_list):
 
 # helper for preview queries
 # tags_id is either a string or a list, which is handled in either case by the len() test. ALL_MEDIA is the exception
-def concatenate_query_for_solr(solr_seed_query, media_ids, tags_ids, custom_ids=[]):
+def concatenate_query_for_solr(solr_seed_query, media_ids, tags_ids, custom_collection=[]):
     query = '({})'.format(solr_seed_query)
 
-    if len(media_ids) > 0 or len(tags_ids) > 0 or len(custom_ids):
+    if len(media_ids) > 0 or len(tags_ids) > 0 or len(custom_collection) > 0:
         if tags_ids == [ALL_MEDIA] or tags_ids == ALL_MEDIA:
             return query
         query += " AND ("
         # add in the media sources they specified
-        if len(media_ids) > 0:
+        if len(media_ids) > 0: # this format is a list of media_idds
             media_ids = media_ids.split(',') if isinstance(media_ids, str) else media_ids
             query_media_ids = " ".join([str(m) for m in media_ids])
             query_media_ids = re.sub(r'\[*\]*', '', str(query_media_ids))
             query_media_ids = " media_id:({})".format(query_media_ids)
             query += '('+query_media_ids+')'
 
         # conjunction
-        if len(media_ids) > 0 and (len(tags_ids) > 0 or len(custom_ids) > 0):
+        if len(media_ids) > 0 and (len(tags_ids) > 0 or len(custom_collection) > 0):
             query += " OR "
 
         # add in the collections they specified
-        if len(tags_ids) > 0:
+        if len(tags_ids) > 0: # this format is a list of tags_id_medias
             tags_ids = tags_ids.split(',') if isinstance(tags_ids, str) else tags_ids
             query_tags_ids = " ".join([str(t) for t in tags_ids])
             query_tags_ids = re.sub(r'\[*\]*', '', str(query_tags_ids))
             query_tags_ids = " tags_id_media:({})".format(query_tags_ids)
-            if len(custom_ids) == 0:
+            if not len(custom_collection) > 0:
```

Comment:
Nit: I'd find this easier to read in the original: `len(custom_ids) == 0`. It took me a second to parse the logic.